### ISSUE TITLE
Exception when using Rails 5.0.0 

### ISSUE THREAD

**Issue by altmer** [dwbutler/logstash-logger#86]
Logstash logger and rails versions:
```
    lograge (0.4.1)
    logstash-event (1.2.02)
    logstash-logger (0.15.1)
    rails (5.0.0)
```

Initialization of logstash logger (in config/application.rb):

```
  config.lograge.enabled = true
  config.lograge.formatter = Lograge::Formatters::Logstash.new
  config.logstash.uri = "file:///var/applications/test-app/log/logstash.log"
```

Fails with following exception:

```
2016-07-22 08:26:25 +0000: Rack app error: #<NoMethodError: super: no superclass method `silence' for #<Logger:0x00000003fc5d98>>
/home/vagrant/.rvm/gems/ruby-2.3.0@global/gems/activesupport-5.0.0/lib/active_support/logger.rb:63:in `block (3 levels) in broadcast'
/home/vagrant/.rvm/gems/ruby-2.3.0@global/gems/activesupport-5.0.0/lib/active_support/logger_silence.rb:20:in `silence'
/home/vagrant/.rvm/gems/ruby-2.3.0@global/gems/activesupport-5.0.0/lib/active_support/logger.rb:61:in `block (2 levels) in broadcast'
/home/vagrant/.rvm/gems/ruby-2.3.0@global/gems/sprockets-rails-3.1.1/lib/sprockets/rails/quiet_assets.rb:11:in `call'
/home/vagrant/.rvm/gems/ruby-2.3.0@global/gems/actionpack-5.0.0/lib/action_dispatch/middleware/request_id.rb:24:in `call'
/home/vagrant/.rvm/gems/ruby-2.3.0@global/gems/rack-2.0.1/lib/rack/method_override.rb:22:in `call'
/home/vagrant/.rvm/gems/ruby-2.3.0@global/gems/rack-2.0.1/lib/rack/runtime.rb:22:in `call'
/home/vagrant/.rvm/gems/ruby-2.3.0@global/gems/activesupport-5.0.0/lib/active_support/cache/strategy/local_cache_middleware.rb:28:in `call'
/home/vagrant/.rvm/gems/ruby-2.3.0@global/gems/actionpack-5.0.0/lib/action_dispatch/middleware/executor.rb:12:in `call'
/home/vagrant/.rvm/gems/ruby-2.3.0@global/gems/actionpack-5.0.0/lib/action_dispatch/middleware/static.rb:136:in `call'
/home/vagrant/.rvm/gems/ruby-2.3.0@global/gems/rack-2.0.1/lib/rack/sendfile.rb:111:in `call'
/home/vagrant/.rvm/gems/ruby-2.3.0@global/gems/railties-5.0.0/lib/rails/engine.rb:522:in `call'
```

The exactly same setup works fine when using Rails 4.2.6. Any ideas on why might this happen?

**Comment by mduqueoviedo** [comment]
Happening to me as well +1

**Comment by joermungandr** [comment]
Oh yeah. +1 Would be nice to find a solution for this

**Comment by dwbutler** [comment]
I agree this needs to be fixed. For now you can try working around the issue by adding `config.assets.quiet = false` to `config/environments/development.rb`.

**Comment by dwbutler** [comment]
After some research, it turns out that ActiveSupport's current logger silencing implementation won't work in custom `Logger` classes because of https://github.com/rails/rails/issues/24263.

I don't think it's a good idea to monkey patch `LogStashLogger` to work around an issue present in some versions of Rails and not others, especially since these seem to be internal implementation details that are bound to change further.

Nor do I want to re-implement the log silencing interface. (Unless this is a feature people want from LogStashLogger independently of making it work with Rails.)

The best solution might be to simply document the issue and workaround in the readme until the issue is fixed upstream in Rails.

**Comment by dwbutler** [comment]
I want LogStashLogger to work out-of-the-box with Rails. So I decided the best thing to do is vendor in a log silencing implementation that will remain stable instead of relying on internal ActiveSupport classes, similar to what was done for tagged logging.

**Comment by dwbutler** [comment]
The fix for this has been released in 0.17.0. No extra configuration is necessary. Logger silencing should "just work".
